tests/contrib/fastapi/test_fastapi.py

```diff
@@ -489,6 +491,56 @@ def test_service_can_be_overridden(client, tracer, test_spans):
     assert span.service == "test-override-service"
 
 
+def test_subapp(client, tracer, test_spans):
+    response = client.get("/sub-app/hello/name")
+    assert response.status_code == 200
+    assert response.json() == {"Greeting": "Hello"}
+
+    spans = test_spans.pop_traces()
+    assert len(spans) == 1
+    assert len(spans[0]) == 3
+    request_span = spans[0][0]
+    assert request_span.service == "fastapi"
+    assert request_span.name == "fastapi.request"
+    assert request_span.resource == "GET /sub-app/hello/{name}"
+    assert request_span.error == 0
+    assert request_span.get_tag("http.method") == "GET"
+    assert request_span.get_tag("http.url") == "http://testserver/sub-app/hello/name"
+    assert request_span.get_tag("http.status_code") == "200"
+    assert request_span.get_tag("http.query.string") is None
+
+
+def test_w_patch_starlette(client, tracer, test_spans):
+
+    patch_starlette()
+
+    response = client.get("/file", headers={"X-Token": "DataDog"})
+    assert response.status_code == 200
+    assert response.text == "Datadog says hello!"
+
+    spans = test_spans.pop_traces()
+    assert len(spans) == 1
+    assert len(spans[0]) == 1
+    request_span = spans[0][0]
+    assert request_span.service == "fastapi"
+    assert request_span.name == "fastapi.request"
+    assert request_span.resource == "GET /file"
+    assert request_span.error == 0
+    assert request_span.get_tag("http.method") == "GET"
+    assert request_span.get_tag("http.url") == "http://testserver/file"
+    assert request_span.get_tag("http.query.string") is None
+    assert request_span.get_tag("http.status_code") == "200"
+    unpatch_starlette()
```

Comment:
Do we want this in a try/finally for unpatching starlette?

if this test fails, it can leak to other test cases.

e.g.

```python

patch_starlette()
try:
    # client.get() + asserts etc
finally:
    unpatch_starlette()
```